Add generic series title

CaleCrawford opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!--- Please make sure to create an issue for your task before creating -->
<!--- a pull pequest. (Ideally, you should create the issue before you -->
<!--- begin work, but better late than never.) This is how we keep track -->
<!--- of the discussion around planned and in-development features and also -->
<!--- how we can later find the code from the discussion and vice versa. -->

<!--- Please complete the following sections with a level of detail -->
<!--- determined at your discretion. In general, the amount of description -->
<!--- should be roughly proportional to the size and significance of the changes. -->

## Description
<!--- Describe your changes in detail. Focus on the high-level "how" rather than -->
<!--- the nitty-gritty "what"; i.e. avoid repeating the diff in prose. -->
Add series tile specifically for podcasts. Depending on everyone's thoughts, we could modify this to be used for all tile layouts. If this is accepted then I will create a list view to wrap these. I added a font helper to fix some font inconsistencies in android as well as a colors file.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please mention the issue here; -->
<!--- e.g. "Closes #4." This also tells GitHub to automatically -->
<!--- close the issue when the PR is merged. -->
Closes #44 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
I did not add tests, I did not see a place for tests yet. I would be interested and willing to implement tests. There were a lot of pieces of the code I was unsure as to the best approach. Please feel free to add any code review comments. I did not update anything in the readme but would be happy to if it seems like that is needed.

## Screenshots (if appropriate):
<!--- Showing off your awesome work is encouraged! -->
<!--- You could also consider a quick demo via animated GIF, -->
<!--- e.g. using https://giphy.com/apps/giphycapture -->
<!--- Alternatively, a picture of a cute animal is always nice. -->
![simulator screen shot - iphone 8 - 2018-02-02 at 21 21 15](https://user-images.githubusercontent.com/5199964/35763718-ed371d7e-0866-11e8-8555-96b265f920fd.png)
![screenshot_1517635140](https://user-images.githubusercontent.com/5199964/35763712-d8cfc868-0866-11e8-910f-eb37ce6c2c8a.png)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] New feature (non-breaking change which adds functionality)
<!--- NOTE: breaking changes will probably be rare until we've amassed some functionality to break. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. -->
- [x] My code follows the code style of this project.

brettdh commented on src/components/PodcastTile.js:
```diff
@@ -0,0 +1,92 @@
+  title: PropTypes.string.isRequired,
+  total_episodes: PropTypes.number.isRequired,
+  last_updated: PropTypes.string.isRequired,
+};
```
IMO `last_updated` should be a timestamp object rather than a string. Then the component decides how to format it as text.

I was thinking of using momentjs for all our timestamps, but I’m open to other suggestions.

brettdh replied:
`PodcastTile` will need an `onPress` prop.

brettdh commented on src/components/PodcastTile.js:
```diff
@@ -0,0 +1,92 @@
+);
+
+PodcastTile.propTypes = {
+  source: PropTypes.shape({
```
`source` is a descriptive name for a property of `Image` but not as descriptive of `PodcastTile`. Maybe `imageSource`?

brettdh commented:
I think you’ve actually almost created the generic `SeriesTile` component and named it `PodcastTile` instead. 😊 If the subtitle was an array of strings instead, that would be the last podcast-specific thing to be pulled out.

brettdh commented:
As far as testing goes - I’m undecided as to whether the mostly-visual components like these merit such tests. Once we are confident they won’t change much, we could consider doing snapshot testing so we know if we accidentally change the look significantly. Also open to suggestions here, though.

CaleCrawford commented:
Yeah it really could. Almost finished with all the other feedback. I’ll make that change and update the PR! Thanks for all the feedback!

CaleCrawford commented:
Okay, I updated the PR and squashed the commits together. 

- I modified the component to accept a react element as a prop that is used as the description. The reason for this is that the podcast description needed more formatting than just a string.

- I added the onPress prop. Not sure what the best approach to this is. I created a function that runs on the parent to do the navigation. If there is a better way to do that I am happy to change it.

- I had some issues with the eslint config when I was trying to add arrow function so I modified the eslint config and then ended up converting it to JS so that we could add comments and have prettier format things.

brettdh commented on src/components/SeriesTile.js:
```diff
@@ -0,0 +1,82 @@
+    uri: PropTypes.string.isRequired,
+  }).isRequired,
+  title: PropTypes.string.isRequired,
+  triggerNavigation: PropTypes.func.isRequired,
```
Nothing about a `SeriesTile` implies a single associated action when it's pressed, so I'd suggest just having an `onPress` here that gets attached to the wrapping `Touchable`.

Also, I've been using `TouchableWithoutFeedback` everywhere so far. I don't know if the absence of touch feedback in the designs is intentional, or a limitation of InDesign, or both - but we should just pick one and be consistent.

Or maybe there are some things that should have feedback and some that shouldn't? I haven't really thought it through, but maybe tiny things like tags should have feedback?

brettdh commented on src/styles/colors.js:
```diff
@@ -0,0 +1,9 @@
+  grey1: '#797979',
+  grey2: '#4A4A4A',
+  grey3: '#333333',
+  grey4: '#1F1F1F',
```
Could we find some more descriptive names for these greys?

Also, it looks like `grey1` is the only one that's actually used... I assume these are just pulled from InDesign? In that case, since we know we want to keep them around, let's either give them names based on what they're used for in the designs, or else leave a comment to do so when we start using them.

brettdh commented on storybook/stories/SeriesTile.stories.js:
```diff
@@ -0,0 +1,59 @@
+            }}
+            title="The Liturgist Podcast"
+            triggerNavigation={this.triggerPress}
+            description={this.formatPodcastDescription('2018-01-19T12:30:18-08:00', 66)}
```
Think we might want to start with a `moment` here rather than a string, since that's the data that will be associated with a podcast.

More meta: I suspect we might want a `PodcastSeriesTile` that renders a `SeriesTile` with API data that's passed as a `podcast` prop. I have a rough example of how this might work for `PersonCard`, in #46 ([source](https://github.com/theliturgists/app/pull/46/files#diff-3b2eabcb7682589f1f9bbc04abf8e2d5R47)).

brettdh commented on src/components/PodcastSeriesDescription.js:
```diff
@@ -0,0 +1,50 @@
+      <View style={styles.descriptionWrapper}>
+        <Text style={styles.subtext}>{this.props.totalEpisodes} Episodes</Text>
+        <Icon
+          style={styles.decriptionIcon}
```
typo: `decription`

brettdh commented:
> I modified the component to accept a react element as a prop that is used as the description. The reason for this is that the podcast description needed more formatting than just a string.

It occurs to me that, if we do in fact make a `PodcastSeriesTile` as I suggested above, it could render `podcast.lastUpdated` (a `moment`) using `PodcastDescription` and pass the result to `SeriesTile` as the `description` prop.

CaleCrawford commented on src/components/PodcastSeriesDescription.js:
```diff
@@ -0,0 +1,50 @@
+
+PodcastDescription.propTypes = {
+  totalEpisodes: PropTypes.number.isRequired,
+  lastUpdated: PropTypes.string.isRequired,
```
Can we have a proptype that validates a `moment`? What does that look like? `PropTypes.number.isRequired`?

brettdh replied:
Something like `PropTypes.instanceOf(moment.moment).isRequired`. I’m not sure if that actually works, but there’s also this:

https://www.npmjs.com/package/react-moment-proptypes
